### ISSUE TITLE
Consider allowing edits to glTF skin definitions

### ISSUE THREAD

**Issue by verma-akash** [google/filament#3075]
Hi, 
Sorry for asking sceneform related bug here but since the repo is archived didn't have other options.
This is related to the sceneform related bug [https://github.com/google-ar/sceneform-android-sdk/issues/1039](https://github.com/google-ar/sceneform-android-sdk/issues/1039)

My question is there a way to directly use filament to find the bones position that are exported in the model and attach a another Model at that position.
Using FilamentAsset from sceneform i was able to check the entity names which included the Bones present in the model, but is there a way to attach nodes to this bone.

**Comment by DylanYasen** [comment]
I've never tried it.  but can you getChildren() and look for your node. and then SetParent() to attach things to it?

**Comment by verma-akash** [comment]
The Bones are not the child nodes of Renderable. So it is not possible. In Sceneform the renderable could be set on a SkeletonNode which provide with the bone information as well and using setAttachment we could connect a different renderable to that bone position.

**Comment by prideout** [comment]
In the gltfio library, the mapping from joint nodes to renderables is only used in [Animator::updateBoneMatrices()](https://github.com/google/filament/blob/main/libs/gltfio/src/Animator.cpp#L349), and yes this mapping is immutable right now, and not exposed to Java clients. (C++ clients can peek at the raw cgltf structs.)

Note that the use of `gltfio::Animator` is optional; if desired you could call `RenderableManager::setBones()` and manually pass in any matrices you wish.

To summarize, this sounds like this is a feature request that would allow SkinVector to be readable and modifiable, both for `gltfio::FilamentInstance` and for `gltfio::FilamentAsset`.

**Comment by prideout** [comment]
@verma-akash are you still blocked by this?  Did my summary make sense?

**Comment by verma-akash** [comment]
@prideout As you mentioned this is a feature request, i'll wait for the update. Thanks

**Comment by dandingol03** [comment]
Hi, @prideout  Is this feature support now?  I also encounter the same problem which @verma-akash mentioned.

**Comment by dandingol03** [comment]
@verma-akash  Recently i need to implement bone-binding function at gltf model, and i come here for some help or insight~

**Here is my scene:**
- There are two gltf model , one is a shoe which will be bind to the role gltf
- These two gltf share the same skeleton and joints
- the shoe gltf which has its own inverseBindMatrices and joint weight

 **Here is my implementation step:**
- i replace the shoe gltf joints' transform with the role's gltf and inverseBindMatrices
- i write a bindOtherBoneMetrics function to make the shoe gltf to get the role's gltf's FFilamentAsset
- i use joints' name to map the shoe's gltf joints to the roles' gltf joints

**Here is my code:**
```
void Animator::bindOtherBoneMatrices(FFilamentAsset* otherAsset) {    
  auto update = [=](const SkinVector& skins, BoneVector& boneVector, utils::NameComponentManager* nameComponentManager, FFilamentAsset* otherAsset = nullptr) {
          for (const auto& skin : skins) {
              size_t njoints = skin.joints.size();
              boneVector.resize(njoints);
              for (const auto& entity : skin.targets) {
  
                  auto renderable = renderableManager->getInstance(entity);
                  if (!renderable) {
                      continue;
                  }
                  std::string name =  nameComponentManager->getName(nameComponentManager->getInstance(entity));
  
                  mat4f inverseGlobalTransform;
                  auto xformable = transformManager->getInstance(entity);
                  if (xformable) {
                      inverseGlobalTransform = inverse(transformManager->getWorldTransform(xformable));
                  }
  
                  for (size_t boneIndex = 0; boneIndex < njoints; ++boneIndex) {
  
                      const auto& joint = skin.joints[boneIndex];
                      TransformManager::Instance jointInstance = transformManager->getInstance(joint);
                      mat4f inverseBindMatrice = skin.inverseBindMatrices[boneIndex];
                      std::string joint_name = nameComponentManager->getName(nameComponentManager->getInstance(joint));
                      Entity other_joint;
                      if (otherAsset->mNameToEntity.count(joint_name) > 0) {
                          other_joint = otherAsset->mNameToEntity[joint_name][0];
                          jointInstance = transformManager->getInstance(other_joint);
                          for (size_t joint_index = 0; joint_index < otherAsset->mSkins[0].joints.size(); joint_index++) {
                              if (otherAsset->mSkins[0].joints[joint_index].getId() == other_joint.getId() ) {
                                  inverseBindMatrice =  otherAsset->mSkins[0].inverseBindMatrices[joint_index];
                                  break;
                              }
                          }
                      }
                      //TransformManager::Instance jointInstance = transformManager->getInstance(other_joint);
                      mat4f globalJointTransform = transformManager->getWorldTransform(jointInstance);
  
                      boneVector[boneIndex] =
                              inverseGlobalTransform *
                              globalJointTransform *
                                      inverseBindMatrice;
                  }
                  renderableManager->setBones(renderable, boneVector.data(), boneVector.size());
              }
          }
      };
  
  
      if (mImpl->instance) {
          update(mImpl->instance->skins, mImpl->boneMatrices, mImpl->asset->mNameManager);
      } else if (!mImpl->asset->isInstanced()) {
          update(mImpl->asset->mSkins, mImpl->boneMatrices, mImpl->asset->mNameManager, otherAsset);
      } else {
          for (FFilamentInstance* instance : mImpl->asset->mInstances) {
              update(instance->skins, mImpl->boneMatrices, mImpl->asset->mNameManager);
          }
      }
  }
}
```
**Problem Description:**
The rendering result seems a little offset
![Screenshot_20220518-183738_gltf viewer](https://user-images.githubusercontent.com/13013579/169024433-27488142-7106-400c-9a0a-157ac661884a.jpg)

**Comment by dandingol03** [comment]
nobody answer this quetion????
